src/Plugin.php

```diff
@@ -35,9 +40,94 @@ class Plugin {
   public static function init() {
+    // Only users with an allowed role ('sale-editor' by default) or with the
+    // 'edit_product' capability can print the price label from the frontend.
+    if (!static::checkUserCanPrintLabel()) {
+      return;
+    }
+
+    if (!Label::$pdfDocsFolderPath || !Label::$pdfDocsFolderUrl) {
+      $uploads_dir = wp_get_upload_dir();
+      Label::$pdfDocsFolderPath = $uploads_dir['basedir'] . '/' . Plugin::PREFIX;
+      Label::$pdfDocsFolderUrl = $uploads_dir['baseurl'] . '/' . Plugin::PREFIX;
+    }
+
+    // Adds price label printing controls to product summary.
+    add_action('woocommerce_simple_add_to_cart', __CLASS__ . '::addPrintPriceLabelControls');
+    add_action('woocommerce_single_variation', __CLASS__ . '::addPrintPriceLabelControls');
+
+    // Prints product price label PDF document.
+    add_action('post_action_label', __NAMESPACE__ . '\Label::post_action_label');
+
+    // Schedule the cleanup of expired products price labels.
+    add_action(Plugin::PREFIX . '_delete_old_price_labels', __NAMESPACE__ . '\Label::deleteOldPriceLabels');
+
     if (is_admin()) {
       return;
     }
+
+    // Enqueues frontend plugin scripts and styles.
+    add_action('wp_enqueue_scripts', __CLASS__ . '::wp_enqueue_scripts');
+  }
+
+  /**
+   * Checks if user is allowed to print product price labels from frontend.
+   *
+   * @return bool
+   *   TRUE if the user can print product price labels from frontend.
+   */
+  public static function checkUserCanPrintLabel() {
+    $user = wp_get_current_user();
+    $allowedRoles = apply_filters(
+      Plugin::PREFIX . '_roles_can_get_price_label',
+      static::ROLES_CAN_GET_PRICE_LABEL
+    );
+
+    // Checks if user has any of the roles allowed to print the label.
+    $matchingRoles = array_filter($user->roles, function ($role) use ($allowedRoles) {
+      return in_array($role, $allowedRoles);
+    });
+
+    return ($user->can('edit_product') || count($matchingRoles));
+  }
+
+  /**
+   * Adds price label printing controls to product summary.
+   *
+   * @implements woocommerce_single_product_summary
+   * @implements woocommerce_single_variation
+   */
+  public static function addPrintPriceLabelControls() {
+    $product_id = get_the_ID();
+    $labelFormatDefault = get_option(Plugin::PREFIX . '-format');
+    Label::addProductPriceLabelControls($product_id, $labelFormatDefault);
+  }
+
+  /**
+   * Enqueues frontend plugin scripts and styles.
+   *
+   * @implements wp_enqueue_scripts
+   */
+  public static function wp_enqueue_scripts() {
+    if (function_exists('is_product') && is_product()) {
+      $git_version = static::getGitVersion();
+      if (defined('SCRIPT_DEBUG') && SCRIPT_DEBUG) {
+        $script = '/assets/scripts/main.js';
+      }
+      else {
+        $script = '/dist/scripts/main.min.js';
+      }
+      // Enqueues styles.
+      wp_enqueue_style(Plugin::PREFIX, static::getBaseUrl() . '/dist/styles/main.min.css', [], $git_version);
```

Comment:
Please see my other comment above about the minification. Also please add minification support for the style as well if you do it for the scripts too.